data-structures/hash-tables.md

```diff
@@ -39,11 +40,25 @@ Knowing how hash tables work will give you a deeper understanding of why hash ta
 
 ### Common Mistakes / Misconceptions
 
-What is the difference between a hash map and a hash table? The two are used interchangeably.
+- What is the difference between a hash map and a hash table? The two are often used interchangeably.
+
+- When should I use an array instead of a hash table? If your keys are sequential integers.
+
+- When does a JavaScript object stop being a hash table? When a property is added as a function.
```

Comment:
Do you mean when a function is added as a property? Or is this correct?

Reply:
I'm not sure about this. Because it was already there. I searched on google but I didn't get any proper answer for this. I am sorry for that. 😞

Reply:
@Kundan28 @manaswinidas @praveenscience @saberprashant Does anyone know the answer?

Reply:
Also, we need an additional review on this PR/ add it to the scoresheet.

Reply:
It could be `When ________ is added as a property of a function.`

Reply:
There is a method called `obj.freeze()` but it is for making the object immutable. So I am not sure about that.

_A frozen object can no longer be changed; freezing an object prevents new properties from being added to it, existing properties from being removed, prevents changing the enumerability, configurability, or writability of existing properties, and prevents the values of existing properties from being changed._

If there is no solution, may I remove it? Because I am trying to find it on google but there are no such results.

Reply:
Try reaching out to people on the slack channels, too. I’m certain that the answer is in the lesson, I remember seeing the difference between an object and a hash table.

Reply:
I think it should be something like,
An object stops being a hash table if we add a property or key as an object or array to it. Because a regular object doesn't accept keys as objects or arrays but a Map does.

I even tried to find out a way, where an object stops being a Map when a function is added as a property, but I am able to use it as a map even after adding the function. Please check this codepen:
https://codepen.io/anon/pen/MdEvGb?editors=0010

And do read this, for finding out the major differences b/w Map & Object:
https://medium.com/front-end-weekly/es6-map-vs-object-what-and-when-b80621932373

Reply:
If no solution is found out..we can rephrase/change that question.

Reply:
I can merge this PR once this is resolved.